subcommand/connect-init/command_test.go

```diff
@@ -105,6 +106,21 @@ func TestRun_ServicePollingWithACLsAndTLS(t *testing.T) {
 			serviceName:        "",
 			expFail:            true,
 		},
+		{
+			name:               "ACLs enabled, k8s-service-name flag is set and matches the service in Consul",
+			tls:                false,
+			serviceAccountName: "counting",
+			serviceName:        "",
+			k8sServiceName:     "counting",
+		},
+		{
+			name:               "ACLs enabled, k8s-service-name flag is set and doesn't match the service in Consul",
+			tls:                false,
+			serviceAccountName: "counting",
+			serviceName:        "",
+			k8sServiceName:     "not-in-consul",
+			expFail:            true,
+		},
```

Comment:
Do we need to add these tests cases to `TestRun_ServicePollingOnly` to test this code without ACLs? I'm thinking that since ACL token affects the kind of services you have access to, it might good to add those too.

Reply:
Good idea, added this!